### DISCUSSION TITLE
Greater than one pixel accuracy for page.get_textbox() method

### DISCUSSION THREAD

**Discussion by coldter** [pymupdf/PyMuPDF#2445] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**

I'm currently working with intently dense PDF and when using get_textbox with rectangle, it returns unwanted surrounding despite having accurate pinpoint coordinates, I believe it's due to limitation of the rectangle only having one pixel accuracy. Is there any way to get a more accurate result?

**Comment by JorjMcKie** [comment]
A typical Discussions item ... transferring

**Reply by JorjMcKie** [reply, marked as answer]
Ultimately, we would need an example file. But as a general comment:
What you describe is likely caused by (1) having a font with a fairly large "natural" line height, and (2) text lines written with a closer distance than this line height.
The is the "fault" of the document creator.
You can identify (locate) your text after first having set a global variable `fitz.TOOLS.set_small_glyph_heights(True)`. This will cause subsequent text searches and extractions use line heights equal to the font size. Which might be sufficient to restrict extractions to the wanted text.

Otherwise, try using an extremely small bbox height (e.g. 20% of the font size). In any case, only characters intersecting the extraction rectangle will be included.
You have all the needed tools in PyMuPDF's tool box, so no need for an enhancement.

But please feel free to provide us with an example underpinning your problem.

**Reply by coldter** [reply]
So my amateur try was something close to this:: https://github.com/coldter/pymupdf_test/blob/main/get_textbox.py

But thanks for explaining that, I'll try to dig more into the docs and the topic...
